### PR TITLE
chore: support symfony 5

### PR DESCRIPTION
**Pull request by SelviA**
* `DataCollector::collect` method signature change since SF 4.4
```php
// before Symfony 4.4
public function collect(Request $request, Response $response, \Exception $exception = null);

// since Symfony 4.4
public function collect(Request $request, Response $response, \Throwable $exception = null);
```
@see https://github.com/symfony/symfony/commit/ffcfdb42a708f91fa9ba201ef43b9af24af27047

* Renamed events :
`GetResponseForExceptionEvent` replaced by `ExceptionEvent`
`PostResponseEvent` replaced by `TerminateEvent`

* Swap arguments of dispatch()
@see symfony/symfony@75369da

* Use the spaceless filter instead of spaceless tag (since Twig 1.38)

### CODE REVIEW

**Comment by SelviA** [comment]
Hi @Oliboy50,

IMO it's a little bit difficult to keep support of Symfony versions before 4.4 but let's try : 

- The signature of the `DataCollector::collect` method has been updated since 4.4 and unless I am mistaken PHP only supports [covariance](https://www.php.net/manual/en/language.oop5.variance.php) since 7.4 version (which is supports since 4.4 version of Symfony)
- The signature of the `EventDispatcherInterface::dispatch()` method has been updated. Maybe we can use the `LegacyEventDispatcherProxy` decorator as suggests [here](https://github.com/symfony/symfony/pull/33115/files) => I can do it if you want :+1: 
- For the renamed events `GetResponseForExceptionEvent` and `PostResponseEvent`, no idea to keep compatibility.


I think it's better to allow recent applications to use this bundle but I am listening to you if you have any suggestions.

File: src/Resources/views/Collector/statsd.html.twig [hunk 1]
```diff
@@ -19,9 +19,9 @@
 
 {% block menu %}
 <span class="label">
-    <span class="icon">{% spaceless %}
+    <span class="icon">{% apply spaceless %}
```
Comment (Oliboy50) [inline src/Resources/views/Collector/statsd.html.twig:22]: please simply remove the `spaceless` tag here and manually remove all the spaces between HTML tags instead 
(using `apply` introduces breaking changes with twig)

```suggestion
    <span class="icon"><img ...
```

File: src/Resources/views/Collector/statsd.html.twig [hunk 2]
```diff
@@ -19,9 +19,9 @@
...
-    {% endspaceless %}</span>
+    {% endapply %}</span>
```
Comment (Oliboy50) [inline src/Resources/views/Collector/statsd.html.twig:24]: same 

```suggestion
    ... alt="Statsd" /></span>
```

File: src/Tests/Units/DependencyInjection/M6WebStatsdExtension.php [hunk 3]
```diff
@@ -19,7 +19,7 @@ class M6WebStatsdExtension extends atoum\test
     protected function initContainer($resource, $debug = false)
     {
         $this->container = new ContainerBuilder();
-        $this->container->register('event_dispatcher', new EventDispatcher());
+        $this->container->register('event_dispatcher', EventDispatcher::class);
```
Comment (Oliboy50) [inline src/Tests/Units/DependencyInjection/M6WebStatsdExtension.php:22]: 🤷

**Comment by SelviA** [comment]
@Oliboy50 Ready for a new release ?

File: .travis.yml [hunk 4]
```diff
@@ -18,3 +18,8 @@ before_script:
 script:
   - bin/coke
   - bin/atoum
+
+jobs:
+  exclude:
+    - php: 7.1
+      env: SYMFONY_VERSION=5.*.*
```
Comment (valentin-claras) [inline .travis.yml:25]: Could you please add a new line at the end of this file :bow:

**Comment by SelviA** [comment]
Are we ready for a new release ?

**Comment by Oliboy50** [comment]
thank you 👌 

=> https://packagist.org/packages/m6web/statsd-bundle#v4.0.0

**Pull request merged**